sweet extensions go.NetId() fix

pyrosshade opened the pull request:
go.NetId() can now retreive matrix netIdentities again
~~Removes unreachable return statement~~

pyrosshade committed:
Fix NetTabs Not displaying

#6603 was almost on the right track

pyrosshade committed:
Merge branch 'unitystation:develop' into Desync-Doctor

pyrosshade committed:
Merge branch 'unitystation:develop' into Desync-Doctor

pyrosshade committed:
remove unreachable code

cannot be a localplayer (client) and headlessserver (client-less server) at the same time

pyrosshade committed:
Fix sweetextensions go.NetId()

go.NetId() now accounts for matrices

Bod9001 commented on UnityProject/Assets/Scripts/Player/PlayerSync.cs:
```diff
@@ -473,10 +473,6 @@ private void UpdateMe()
 
 	private void Synchronize()
 	{
-		if (isLocalPlayer && GameData.IsHeadlessServer)
```
Technically this can trigger on the headless server,m idk how it handles its own playerr

pyrosshade replied:
I'm under the impression that headless is server only? There shouldn't be a localplayer for something that is not a client.

ThatDan123 replied:
Yeah, i dont think this check was ever true. So removing it wouldnt change anything. But just making it a Headless check (if that was the original intension) could break stuff, so would need testing.

ThatDan123 replied:
![image](https://user-images.githubusercontent.com/56727168/120243227-f64eb600-c25e-11eb-85a3-74612d2a66aa.png)

i donno maybe we should keep it in. Might be some weird interaction with headless

corp-0 commented on UnityProject/Assets/Scripts/Util/SweetExtensions.cs:
```diff
@@ -82,7 +82,11 @@ public static T GetRandom<T>(this List<T> list)
 	public static uint NetId(this GameObject go)
 	{
-		return go ? go.GetComponent<NetworkIdentity>().netId : global::NetId.Invalid; //maxValue is invalid (see NetId.cs)
+		return go == null
```
I like ternaries but in this case the nested ternaries might be hard to read for people with less experience. I'd suggest using a simple if statement in this case.

ThatDan123 commented on UnityProject/Assets/Scripts/Util/SweetExtensions.cs:
```diff
@@ -82,7 +82,11 @@ public static T GetRandom<T>(this List<T> list)
+		return go == null
+			? global::NetId.Invalid
+			: go.GetComponent<Matrix>()
+				? go.GetComponentInChildren<NetworkIdentity>().netId 
```
Splitting it out into If statements would be better like Gilles said and then do TryGetComponent<Matrix>(out var matrix)

and then returning matrix.NetworkedMatrix.MatrixSync.netId

pyrosshade committed:
revert PlayerSync

until further testing with headless can confirm removing the check at the start of Synchronize() does not create unintended behavior, this will be returned to normal

pyrosshade committed:
go.NetId() is hopefully more readable now

pyrosshade committed:
Merge branch 'Desync-Doctor' of https://github.com/pyrosshade/unitystation into Desync-Doctor
